packages/@okta/vuepress-site/docs/guides/pwd-optional-new-sign-up-email/main/aspnet/integrationsteps.md

```diff
@@ -0,0 +1,91 @@
+
+```csharp
+var userProfile = new UserProfile();
+userProfile.SetProperty("firstName", model.FirstName);
+userProfile.SetProperty("lastName", model.LastName);
+userProfile.SetProperty("email", model.Email);
+var registerResponse = await _idxClient.RegisterAsync(userProfile);
+```
+
+### 4. The user verifies their identity using the email authenticator
+
+Query the `AuthenticationStatus` property of the `AuthenticationResponse` object returned by `RegisterAsync()` to discover the current status of the authentication process. If you've configured your Okta org correctly, the status is `AwaitingAuthenticatorEnrollment`, indicating that the user needs to verify their identity with the email authenticator challenge.
+
+```csharp
+if (registerResponse.AuthenticationStatus ==
+      AuthenticationStatus.AwaitingAuthenticatorEnrollment)
+{
+   Session["idxContext"] = registerResponse.IdxContext;
+   Session["authenticators"] = ViewModelHelper
+      .ConvertToAuthenticatorViewModelList(registerResponse.Authenticators);
+   return RedirectToAction("SelectAuthenticator", "Manage");
+}
+
+ModelState.AddModelError(string.Empty, $"Oops! Something went wrong.");
+return View("Register", model);
+```
+
+The email authenticator supports user verification by One-Time Password (OTP) and by magic links. To learn more, see the [Okta email integration guide](/docs/guides/authenticators-okta-email/nodeexpress/main/#integrate-email-challenge-with-magic-links).
+
+### 5. Your app displays the remaining optional authenticators
+
+After the user verifies their identity using the email authenticator, the `AuthenticationStatus` property of the `AuthenticationResponse` object is either:
+
+- `Success`: Go to step six.
+- `AwaitingAuthenticatorEnrollment`: The user can enroll in additional authenticators.
+
+If the latter is true, create and display a page that lists the remaining authenticators. Check the `CanSkip` property of the `AuthenticationResponse` object. If `true` &mdash; and all the listed authenticators are optional &mdash; add a **Skip** button to the form to skip their enrollment. If `CanSkip` is `false`, you should omit the **Skip** button.
+
+<div class="half border">
+
+![A form showing a list of authenticators to enroll, and Submit, Cancel, and Skip buttons.](/img/pwd-optional/pwd-optional-sign-up-authenticators-page.png)
+
+</div>
+
+### 6. The user skips the remaining optional authenticators
+
+When the user clicks the **Skip** button, call `IdxClient.SkipAuthenticationSelectionAsync()` passing in the `IdxContext` that represents the current state of the registration flow. The `AuthenticationStatus` property of the `AuthenticationResponse` object returned by `SkipAuthenticationSelectionAsync()` is `Success`. You can now call `AuthenticationHelper.GetIdentityFromTokenResponseAsync()` to retrieve the OIDC claims information about the user and pass them into your application. The user has now signed in.
```

Comment:
Suggestion for a somewhat wordy sentence:

Current:
The `AuthenticationStatus` property of the `AuthenticationResponse` object returned by `SkipAuthenticationSelectionAsync()` is `Success`. 

New:
 `SkipAuthenticationSelectionAsync()` returns `AuthenticationResponse.AuthenticationStatus` = `Success`.